app/Services/InboundMail/InboundMailEngine.php

```diff
@@ -0,0 +1,315 @@
+<?php
+/**
+ * Invoice Ninja (https://invoiceninja.com).
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2023. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+namespace App\Services\InboundMail;
+
+use App\Factory\ExpenseFactory;
+use App\Jobs\Util\SystemLogger;
+use App\Libraries\MultiDB;
+use App\Models\ClientContact;
+use App\Models\Company;
+use App\Models\SystemLog;
+use App\Models\VendorContact;
+use App\Services\EDocument\Imports\ParseEDocument;
+use App\Services\InboundMail\InboundMail;
+use App\Utils\TempFile;
+use App\Utils\Traits\GeneratesCounter;
+use App\Utils\Traits\SavesDocuments;
+use App\Utils\Traits\MakesHash;
+use Cache;
+use Illuminate\Queue\SerializesModels;
+
+class InboundMailEngine
+{
+    use SerializesModels, MakesHash;
+    use GeneratesCounter, SavesDocuments;
+
+    private ?bool $isUnknownRecipent = null;
+    private array $globalBlacklist = explode(",", config('global_inbound_blocklist'));
+    private array $globalWhitelist = explode(",", config('global_inbound_whitelist')); // only for global validation, not for allowing to send something into the company, should be used to disabled blocking for mass-senders
+    public function __construct()
```

Comment:
Does this work? I didn't this it was possible to init a property like this unless it was a constant or type?

Reply:
it is a constant, but i will recheck this exspecialy

Reply:
you were right, i fixed it accordingly